e-commerce/shop-backend/src/operations.ts

```diff
@@ -237,6 +323,10 @@ export class Shop {
 
   @Transaction()
   static async subtractInventory(ctxt: KnexTransactionContext, products: Product[]): Promise<void> {
+    return await Shop.subtractInventoryInternal(ctxt, products);
+  }
+
+  static async subtractInventoryInternal(ctxt: KnexTransactionContext, products: Product[]): Promise<void> {
```

Comment:
Why do we need this pass-through function?

Reply:
Jest spyOn is currently not working with DBOS wrapped functions and the test wants to intercept it, so I broke it out as `subtractInventoryInternal` for the spy... see line 185 of the test.

Reply:
I spent a few hours the other week trying to fix `jest` - have our handler/wrapper go find the current function assigned to the class (as it could be a spy now) and invoke that... but all the incantations I tried failed to achieve that.  Open to ideas.